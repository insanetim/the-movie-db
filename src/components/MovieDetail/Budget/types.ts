import { IMovieDetail } from 'src/interfaces/movie.interface'

export interface BudgetProps {
  budget: IMovieDetail['budget']
}
