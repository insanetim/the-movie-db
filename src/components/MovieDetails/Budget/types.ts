import { IMovieDetails } from 'src/interfaces/movie.interface'

export type BudgetProps = {
  budget: IMovieDetails['budget']
}
