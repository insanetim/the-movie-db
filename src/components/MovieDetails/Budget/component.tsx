import { Col, Typography } from 'antd'
import convertMoney from 'src/utils/convertDataHelpers/convertMoney'

import { BudgetProps } from './types'

const Budget: React.FC<BudgetProps> = ({ budget }) => (
  <Col span={24}>
    <Typography.Paragraph>
      <b>Budget: </b>
      <span>{convertMoney(budget)}</span>
    </Typography.Paragraph>
  </Col>
)

export default Budget
