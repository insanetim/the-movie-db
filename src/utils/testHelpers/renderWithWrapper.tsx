import { render, renderHook, RenderOptions } from '@testing-library/react'
import { PropsWithChildren, ReactNode, Suspense } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { AppStore, RootState, setupStore } from 'src/store'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: RootState
  store?: AppStore
}

const createWrapper =
  (store: AppStore) =>
  ({ children }: PropsWithChildren) => (
    <Provider store={store}>
      <Suspense fallback={null}>
        <BrowserRouter>{children}</BrowserRouter>
      </Suspense>
    </Provider>
  )

export const renderWithWrapper = (
  ui: ReactNode,
  {
    preloadedState = {} as RootState,
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = createWrapper(store)
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export const renderHookWithWrapper = <T,>(
  hook: () => T,
  {
    preloadedState = {} as RootState,
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = createWrapper(store)
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
