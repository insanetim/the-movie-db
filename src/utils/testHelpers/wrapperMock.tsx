import { ReactNode } from 'react'
import { BrowserRouter } from 'react-router-dom'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <BrowserRouter>{children}</BrowserRouter>
}
export default Wrapper
