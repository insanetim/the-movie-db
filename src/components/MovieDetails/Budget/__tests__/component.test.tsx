import { renderWithWrapper } from 'src/utils/testHelpers/renderWithWrapper'

import Budget from '../component'

describe('Budget component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<Budget budget={1000000} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
