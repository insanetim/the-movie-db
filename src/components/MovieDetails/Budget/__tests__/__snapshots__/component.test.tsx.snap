// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-24 css-dev-only-do-not-override-5wsri9"
  >
    <div
      class="ant-typography css-dev-only-do-not-override-5wsri9"
    >
      <b>
        Budget: 
      </b>
      <span>
        $1,000,000
      </span>
    </div>
  </div>
</DocumentFragment>
`;
